specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json

```diff
@@ -0,0 +1,302 @@
+      "get": {
+        "operationId": "Offers_Get",
+        "description": "Get Offer resource details within a publisher of HCI Cluster.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/PublisherNameParameter"
+          },
+          {
+            "$ref": "#/parameters/OfferNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ODataExpandParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Offer"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Offer": {
+            "$ref": "./examples/GetOffer.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "OfferList": {
+      "description": "List of Offer proxy resources for the HCI cluster.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "List of Offer proxy resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Offer"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to the next set of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "Offer": {
+      "description": "Offer details.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Offer properties.",
+          "$ref": "#/definitions/OfferProperties",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "OfferProperties": {
+      "description": "Publisher properties.",
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "description": "Provisioning State",
+          "type": "string",
+          "readOnly": true
+        },
+        "publisherId": {
```

Comment:
>publisherId [](http://example.com/codeflow?start=9&length=11)

It helps to describe what kind of Id this is. If it's an ARM resource, then "publisherResourceId" would be recommended (and updated in the description). If it's a different kind of Id, then adding some info in the description would be helpful.

Reply:
Description already added

Reply:
But it's ambiguous. If it's not an ARM resource Id, then what is it? A guid? A user-selected name, something else? It's helpful for users of your APIs to have more than "publisher Id" to go on.

Reply:
PublisherId is one of the attributes to determine marketplace image offer. A marketplace image in azure has publisher, offer, sku and version attributes which uniquely identify a given image offer.
We tried to align with existing compute RP marketplace APIs, so we followed similar naming convention for publishers, offers and skus resource types in our RP as well.